include/globals.js

```diff
@@ -802,3 +820,69 @@ class ContentResolver {
       }
   }
 }
+
+function HandleTSHStateUpdateMessage(message) {
```

Comment:
This is the only function with new-ish logic since I had to rip out React state-update stuff and do normal data manipulation instead.